addons/calendar/static/tests/activity_tests.js

```diff
@@ -35,3 +35,42 @@ QUnit.test("activity click on Reschedule", async () => {
     await click(".btn", { text: "Reschedule" });
     await contains(".o_calendar_view");
 });
+
+QUnit.test("activity with event click on cancel", async (assert) => {
+    const pyEnv = await startServer();
+    const partnerId = pyEnv["res.partner"].create({});
+    const activityTypeId = pyEnv["mail.activity.type"].create({
+        icon: "fa-calendar",
+        name: "Meeting",
+    });
+    const attendeeId = pyEnv["calendar.attendee"].create({
+        partner_id: partnerId,
+    });
+    const calendaMeetingId = pyEnv["calendar.event"].create({
+        res_model: "calendar.event",
+        name: "meeting1",
+        start: "2022-07-06 06:30:00",
+        attendee_ids: [attendeeId],
+    });
+    const activityId = pyEnv["mail.activity"].create({
+        name: "Small Meeting",
+        activity_type_id: activityTypeId,
+        can_write: true,
+        res_id: partnerId,
+        res_model: "res.partner",
+        calendar_event_id: calendaMeetingId,
+    });
+    const { openFormView } = await start({
+        async mockRPC(route, args) {
+            if (args.method === "unlink_w_meeting") {
+                assert.step("unlink_w_meeting");
+                assert.strictEqual(args.args[0].length, 1);
+                assert.strictEqual(args.args[0][0], activityId);
+            }
```

Comment:
This `mockRPC` seems useless since you're checking that the activity disappeared. We know for a fact this method was called for the correct activity, don't we?